prodsys/factories/resource_factory.py

```diff
@@ -163,16 +167,18 @@ def add_resource(self, resource_data: RESOURCE_DATA_UNION):
             name=resource_data.control_policy, cls_dict=CONTROL_POLICY_DICT
         )
         controller: Union[
-            control.ProductionController, control.TransportController
+            control.ProductionController, control.TransportController, control.BatchController
         ] = controller_class(control_policy=control_policy, env=self.env)
         self.controllers.append(controller)
         values.update({"controller": controller})
 
         if isinstance(resource_data, ProductionResourceData):
             input_queues, output_queues = self.get_queues_for_resource(resource_data)
             values.update(
-                {"input_queues": input_queues, "output_queues": output_queues}
+            {"input_queues": input_queues, "output_queues": output_queues}
             )
+            if "batch_size" in resource_data.dict():
```

Comment:
I think this extra bit is also not necessary if the batch size equals the capacity.